docs: Config overview improvement (#28209)
renovatebot/renovate: docs/usage/config-overview.md

When Renovate runs on a repository, the final config used is derived from the:

- Default config (bot defaults)
- Global config (admin settings)
- Inherited config (org-level settings)

> - Repository config (repo-specific customisation)

Reviewer: ```suggestion
- Repository config (repo-specific customization)
```

We use American English, so this is with a z. 🙂

### Global config

Global config means: the config defined by the person or team responsible for running the bot.
This is also referred to as "bot config", because it's the config passed to the bot by the administrator that runs the bot.
Global config can contain config which is "global only" as well as any configuration options which are valid in Inherited config or Repository config.

> Global config must be supplied when you run renovate locally or "self-host" renovate.

Reviewer: ```suggestion
Global config must be supplied when you run Renovate locally or "self-host" Renovate.
```

Capitalize proper nouns.

Maybe this sentence needs to be moved down a bit? We mention below how users of the Mend hosted app don't need to worry about global config. But this sentence almost sounds like all users must configure global config.

But if you have configured `RENOVATE_CONFIG_FILE` and the path you specified is not found then Renovate will error and exit, because it assumes you have a configuration problem.
If the file is found but cannot be parsed then Renovate will also error and exit.

> Global config files can be `.js` or `.json` or `.yaml` files, and must accordingly be formatted using JavaScript, JSON or YAML syntax.

Reviewer: ```suggestion
Global config files can be `.js`, `.json` or `.yaml` files.
You must use the correct syntax for the file format, so JavaScript, JSON or YAML.
```

You may use synchronous or asynchronous methods inside a `.js` file, including even to fetch config information from remote hosts.

#### Environment config

> Most config options can be set using environment variables. 

Reviewer: ```suggestion
Most config options can be set using environment variables.
```

The environment variable name is usually found by adding the prefix `RENOVATE_` to the config option name in all-uppercase.

> (This prefix can be changed by setting the environment variable `ENV_PREFIX`.) 

Reviewer: ```suggestion
(This prefix can be changed by setting the environment variable `ENV_PREFIX`.)
```

For example, `RENOVATE_PLATFORM=gitlab` is the same as setting `"platform": "gitlab"` in File config.

As some environment variable names differ from the usual pattern, we recommend you still check the `env` field of the documentation for each config option to make sure.

> If the configuration option lacks the `env` field, that config option may lack the facility to be configured by a separate environment variable.

Reviewer: ```suggestion
If the configuration option does not have the `env` field, that config option can not be configured by a separate environment variable.
```

I think something like this is easier to understand?

The values assigned to the environment variables may be strings, numbers, booleans, or stringified JSON.

> If using stringified JSON then be careful to use appropriate escaping.

Reviewer: ```suggestion
When you use stringified JSON check if you're escaping it properly.
```

Key names should be quoted using double quotes, as these are strings in JSON.

> If environment variables are being set from a shell console or shell script, it is generally simplest to wrap the entire stringified JSON in single quotes to prevent the shell interpreter from parsing (and removing) the escaping or punctuation.

Reviewer: ```suggestion
If environment variables are being set from a shell console or shell script: wrap the entire stringified JSON in _single_ quotes.
This prevents the shell interpreter from parsing (and removing) the escaping or punctuation.
```

> If the environment variables are being set from a file (such as a docker env-file or in a Kubernetes manifest YAML) then those outermost single-quotes should usually be removed, otherwise renovate may interpret the entire value as a simple string instead of as a compound JSON object.

Reviewer: ```suggestion
If the environment variables are being set from a file (such as a Docker env-file or in a Kubernetes manifest YAML) then those _outermost_ single-quotes should be removed.
This prevents Renovate from wrongly interpreting the entire value as a simple string.
```
